src/modules/auth/components/CustomDataInput.js

```diff
@@ -0,0 +1,67 @@
+import {useTranslation} from "react-i18next";
+import {Button, FormFeedback, FormGroup, Input, InputGroup, InputGroupAddon, Label} from "reactstrap";
+import {Controller} from "react-hook-form";
+import React from "react";
+
+export default function CustomDataInput ({name, control, errors, append, remove, fields, replace, labelName}) {
+	const { t } = useTranslation();
+
+	return (
+		<FormGroup>
+			{labelName && <Label for={name} title={name}>{labelName}</Label>}
+			{fields && fields.map((item, idx) => {
+				if (fields[idx].key === "undefined") {
+					return
+				}
+				return (
+					<InputGroup key={item.id} className="mb-1 custom-data" >
+						<div className="custom-data-key" >
+							<Controller
+								render={({field}) => <Input {...field} invalid={errors[name]?.[idx]?.key}/>}
+								name={`${name}[${idx}].key`}
+								control={control}
+								rules={{
+									validate: {
+										val: value => (/^[a-zA-Z][a-zA-Z0-9_-]{0,126}[a-zA-Z0-9]$|^$/).test(value) || t("CustomDataContainer|Invalid format"),
+									}
+								}}
+							/>
+							{errors[name]?.[idx]?.key && <FormFeedback>{errors[name]?.[idx]?.key?.message}</FormFeedback>}
+						</div>
+						<div className="custom-data-value">
+							<Controller
+								render={({field}) => <Input {...field} invalid={errors[name]?.[idx]?.value}/>}
+								name={`${name}[${idx}].value`}
+								control={control}
+							/>
+						</div>
+						<InputGroupAddon addonType="append" className="custom-data-remove-button" >
+							<Button
+								key={idx}
+								title={(fields.length === 1) && (fields[0].key === "") ? t("CustomDataContainer|Nothing to remove") : t("CustomDataContainer|Remove input")}
+								color="danger"
+								outline
+								size="sm"
+								disabled={(fields.length === 1) && ((fields[0].key === ""))}
```

Comment:
@Aringoaway What are you checking with this condition?

Reply:
@Pe5h4 This condition checks if the number of inputs is 1 and nothing has been added to the input, then the button will be disabled. If text is added to the input, the disabled button becomes a normal, or if will be added next input, the disabled button also becomes a normal. 

https://user-images.githubusercontent.com/65866093/227138162-dc4b1af2-9a1f-456c-9970-60489b25b87d.mp4

Reply:
Ok, add the description then please